strax/corrections.py

```diff
@@ -0,0 +1,127 @@
+'''I/O format for corrections using MongoDB
+
+This module list, reads, writes, among others functionalities
+to a MongoDB where corrections are stored.
+'''
+
+from datetime import datetime
+import logging
+import pandas as pd
+import pdmongo as pdm
+import pymongo
+
+import strax
+
+export, __all__ = strax.exporter()
+
+
+@export
+class CorrectionsInterface:
+    '''A class to manage corrections that are stored in a MongoDB
+    corrections are defined as pandas DataFrame with a DatetimeIndex
+    with the option of a Global configuration, meaning a unique set 
+    of correction maps.
+    '''
+
+    def __init__(self, host='127.0 0.1', username=None, password=None,
+                 database_name='corrections'):
+
+        self.host = host
+        self.username = username
+        self.password = password
+        self.client = pymongo.MongoClient(host=self.host,
+                                          username=self.username,
+                                          password=self.password)
+        self.database_name = database_name
+        self.database = self.client[self.database_name]
+
+    def list_corrections(self):
+        '''Smart logic to list all corrections.
+        '''
+        return [x['name'] for x in self.database.list_collections()]
+
+    def read(self, correction):
+        '''Smart logic to read corrections,
+        :param correction: correction's name (str type).
+        '''
+        df = pdm.read_mongo(correction, [], self.database)
+
+        # No data found
+        if df.size == 0:
+            return None
+        # Delete internal Mongo identifier
+        del df['_id']
+        return df.set_index('time')
+
+    def interpolate(self, what, when, how='interpolate'):
```

Comment:
done

Reply:
Sorry, that I had to unresolve this one, but this will throw an error since the kwargs of the two functions are not the same MWE:

```
kwargs = dict(method='linear')
df.interpolate(**kwargs)   # works
df.ffill(**kwargs) # fails and throws an error
```

Reply:
@WenzDaniel , you also don't want to use those kwargs for ffill? I don't see why it should be accepting kwargs that it's not meant for. You use either of the two methods not both.

Reply:
Ah yes, sorry just saw your example did not have a look into the code itself